eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```diff
@@ -28,61 +29,73 @@ parameters:
+      pool:
+        vmImage: $(OSVmImage)
 
-    pool:
-      vmImage: $(OSVmImage)
+      steps:
+        - template: /eng/common/pipelines/templates/steps/bypass-local-dns.yml
 
-    steps:
-      - template: /eng/common/pipelines/templates/steps/bypass-local-dns.yml
+        - ${{ if eq(parameters['AllocateResourceGroup'], 'true') }}:
+          - pwsh: |
+              $BaseName = 't' + (New-Guid).ToString('n').Substring(0, 16)
+              $RgName = "rg-${{ parameters.ServiceDirectory }}-$($BaseName)"
+              $ResourceGroupExpiration = (Get-Date).ToUniversalTime().AddHours(24).ToString("o")
+              Write-Host "Creating Resource Group: $($RgName)"
+              Write-Host "##vso[task.setvariable variable=AZURE_RESOURCEGROUP_NAME]$RgName"
 
-      - ${{ if eq(parameters['AllocateResourceGroup'], 'true') }}:
-        - pwsh: |
-            $BaseName = 't' + (New-Guid).ToString('n').Substring(0, 16)
-            $RgName = "rg-${{ parameters.ServiceDirectory }}-$($BaseName)"
-            $ResourceGroupExpiration = (Get-Date).ToUniversalTime().AddHours(24).ToString("o")
-            Write-Host "Creating Resource Group: $($RgName)"
-            Write-Host "##vso[task.setvariable variable=AZURE_RESOURCEGROUP_NAME]$RgName"
+              az login --service-principal -u $(provisioner-aad-id) -p $(provisioner-aad-secret) --tenant $(provisioner-aad-tenant)
+              az account set --subscription $(provisioner-subscription)
 
-            az login --service-principal -u $(provisioner-aad-id) -p $(provisioner-aad-secret) --tenant $(provisioner-aad-tenant)
-            az account set --subscription $(provisioner-subscription)
+              az group create --location westus2 --name $RgName --tags DeleteAfter=$ResourceGroupExpiration BuildId='$(Build.BuildId)' BuildJob='$(Agent.JobName)' BuildNumber='$(Build.BuildNumber)' BuildReason='$(Build.Reason)'
+              az group wait --created --resource-group $RgName
+            displayName: Create resource group and environment variable for tests
 
-            az group create --location westus2 --name $RgName --tags DeleteAfter=$ResourceGroupExpiration BuildId='$(Build.BuildId)' BuildJob='$(Agent.JobName)' BuildNumber='$(Build.BuildNumber)' BuildReason='$(Build.Reason)'
-            az group wait --created --resource-group $RgName
-          displayName: Create resource group and environment variable for tests
+          # figure out what's up with the create resource group above
+        - template: ../../../common/TestResources/deploy-test-resources.yml
+          parameters:
+            Location: ${{ coalesce(parameters.Location, cloudConfig.value.Location, 'westus2') }}
+            ServiceDirectory: '${{ parameters.ServiceDirectory }}'
+            SubscriptionConfiguration: ${{ cloudConfig.value.SubscriptionConfiguration }}
+            ArmTemplateParameters: ${{ cloudConfig.value.ArmTemplateParameters }}
+            AdditionalParameters: $ {{ parameters.AdditionalTestArgs }}
```

Comment:
These will not get used. AdditionalTestArgs, in other language pipelines, is passed to the test runner cli.